dpctl/tensor/_statistical_functions.py

```diff
@@ -144,17 +141,18 @@ def _var_impl(x, axis, correction, keepdims):
     res_shape = res.shape
     # when nelems - correction <= 0, yield nans
     div = max(nelems - correction, 0)
-    if not div:
-        div = dpt.nan
-    div_ary = dpt.asarray(
-        div, dtype=res_dt, usm_type=res_usm_type, sycl_queue=q
-    )
-    # divide in-place again
-    if div_ary.shape != res_shape:
-        div_ary = dpt.broadcast_to(div_ary, res.shape)
+    if div:
+        dep_evs = _manager.submitted_events
+        ht_e7, d_e2 = tei._divide_by_scalar(
+            src=res, scalar=div, dst=res, sycl_queue=q, depends=dep_evs
+        )
+        _manager.add_event_pair(ht_e7, d_e2)
+        return res, [d_e2]
+
+    div = dpt.nan
```

Comment:
```suggestion
    if not div:
        div = dpt.nan
```

I made the change originally because `nelems > corrections` branch used divide by scalar, and the else branch used divide in-place. 

Later, both branches were changed to use divide by scalar, but I should have also changed the branching logic.